docs/architecture/adr-104-out-of-band-state-sync.md

```diff
@@ -124,58 +138,173 @@ in order to adopt this solution.
-   running node (server side) to the new node (client side) to be bootstrapped
-   using this new state sync feature.
+### Application-implemented local state sync
 
+This section describes the solution to local state sync implemented by the SDK. An application can 
+chose to implement local state sync differently, or implement only a subset of the functionalities
+implemented by the SDK.
 
-## Detailed Design
+This solution exposes a command line interface enabling a node to manipulate the snapshots
+including dumping existing snapshots to an exportable format, loading and deleting exported snapshots,
+as well as bootstraping a node's state based on a local snapshot. 
+
+The SDK exposes the following commands for snapshot manipulation:
+
+```script
+
+delete      Delete a local snapshot
+dump        Dump the snapshot as portable archive format
+export      Export app state to snapshot store
+list        List local snapshots
+load        Load a snapshot archive file into snapshot store
+restore     Restore app state from local snapshot
+
+```
+
+and the following command to bootstrap the state of CometBFT upon a completed state sync:
+
+``` script
+
+comet bootstrap-state          Bootstrap the state of CometBFT's state and block store
+
+```
+
+These commands enable the implementation of both the client and server side of statesync. 
+Namely, a statesync server can use `dump` to create a portable archieve format out existing snapshots,
+or trigger snapshot creation using `export`. 
+
+The client side, restores the application state from a local snapshot that was previously exported, using
+`restore`. Upon successful completion of the previous sequence of commands, the state of CometBFT is bootstrapped
+using `bootstrap-state` and CometBFT can be launched. 
+
+
+There are three prerequisites for this solution to work when a node is syncing:
+1. The application has access to the snapshot database
+2. CometBFTs state and block stores are empty or reset
+3. CometBFT is not running while the node is state syncing 
+
+The server side of state sync (snapshot generation and dumping), can be performed while the node is running.
+The application has to be careful not to interfere with normal node operations, and to use a snapshot store
+and dumping mechanism that will mitigate the risk of requesting snapshots while they are being dumped to an archive format. 
+
+In order to be able to dump or export the snapshots, the application must have access to the snapshot store. 
+
+We describe the main interface expected from the snapshot database and used by the above mentioned CLI commands
+for snapshot manipulation. 
+The interface was derived from the SDK's implementation of local state sync.  
+
+```golang
+
+// Delete a snapshot for a certain height
+func (s *Store) Delete(height uint64, format uint32) error
+
+// Retrieves a snapshot for a certain height
+func (s *Store) Get(height uint64, format uint32) (*types.Snapshot, error) 
```

Comment:
`types` here is ABCI types?

Reply:
I think here it is SDK specific, I had it on my todo to verify the types and make sure they are Comet consistent.